submissions/balancer-design/universal_8to8.md

```diff
@@ -0,0 +1,68 @@
+
+* Input-balanced: Draws from all inputs evenly. Especially important when merging many inputs to a few outputs.
+* Output-balanced: Gives to all outputs evenly. Especially important when splitting a few inputs to many outputs.
+* Throughput-limited: Has an input-output configuration that creates an internal bottleneck. For example, merging two belts into one and splitting back to two.
+* Throughput-unlimited: Opposite of throughput-limited.
+
+All blueprints from this article (and a couple extras) can be found at [this link](https://controlc.com/c21ae405).
+
+### The premise
+
+Picture the following: you're a new player, and you've just heard about those things called "balancers" and how useful they are. You see an image of a 4-4 balancer, build it, but then only hook up 3 outputs to get 3 balanced belts. Instead you're met with a 2:1:1 ratio and disappointment:
+
+![How not to use a balancer](https://cdn.discordapp.com/attachments/699618487097884723/813449477028446269/unknown.png)
+
+In this article, I set myself a goal to make a balancer for which this sort of intuition works, in other words, no matter the input/output configuration, all outputs are always equal, and all inputs are drawn from evenly. This kind of balancer is known as a "universal balancer", a concept first explored and named in [this post](https://www.reddit.com/r/factorio/comments/a5ferf/i_present_to_you_the_44_universal_balancer/) by Reddit user [u/tzwaan](https://www.reddit.com/user/tzwaan), who is one of the moderators of the [Factorio subreddit](https://www.reddit.com/r/factorio).
+
+Universal balancers can do what no balancer should be able to: no matter the configuration, they're input-balanced, output-balanced, *and* throughput-unlimited at the same time! This is a set of properties previously thought to be possessed only by a 2-2 balancer, which is just a single splitter. The drawback is that they get *big*.
+
+### The design process
+
+Let's think about how to convert a 1-4 balancer into a 1-3. That is done by looping the extra output back into the first splitter. The same can be done with a 4-4 balancer, which we will use as our base principle:
+
+![Looped 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813449600852819968/unknown.png)
+
+Now, how would we make it so the balancing doesn't get ruined when we block some outputs? Simple, we use priority splitters to auto-loop the items that can't get out. Of course, we need to merge them back with the input too, and set input priority so the main inputs don't slow down when there are items coming:
```

Comment:
This step is a little unclear. Do you attempt to only loop back the overflow items for each belt separately?  But the loop -back can only be merged in again if the input belt is not full already? It seems like the belts do this, but I think it should be said explicitly as well.